blog/_posts/2020-06-03-systematics-blogpost.md

```diff
@@ -0,0 +1,335 @@
+
+```c++
+int chooseOrg(vector<Organism> &currentGen, emp::Random &randNum){
+
+    parentNum = randNum.GetInt(size(currentGen));  //chooses random spot 
+    return parentNum;
+}
+```
+
+Here the Empirical random number generator was utilized to ensure that results were actually random. First, a random number was generated based on the size of the parent generation. That spot in the array was then set as the parent of the next generation. 
+
+In the null model, each time a new organism was created it represented its own clade or taxon to ensure maximum diversity. 
+
+### **Mutation and Pressure for Diversity** 
+
+**Mutation**
+
+The trees we used for comparison were trees with mutations and pressure for diversity. 
+
+The mutation rate used for all of the trees was 0.05, which is a typical value for tree modeling in Empirical. In these models, each organism had a genotype as an attribute. 
+
+Mutation was determined randomly. The population generated in the first round of the tree all had a genotype of integer 0. A random double between 0 and 1 was generated with each creation of an organism following this generation. If the value generated was less than 0.05, the genotype would mutate. If a mutation was required, a new random number would be generated between -3 and 3. That genotype would then be subtracted from the original genotype. 
+
+For example, if the organism had a genotype of 2, and was chosen to mutate, and the mutated genotype generated was -3, the new genotype for that organism would be 2 - (-3), which is 5. 
+
+Mutations are also heritable, meaning that the child of an organism would inherit the same mutated or not mutated genotype as its parent. Once an organism mutated, it would create a branch in the tree. 
+
+The following code shows how the organism class handles mutations. In the model that just used mutations but did not account for any pressure to diversify, this code is used as the organism class, but has no fitness calculations and still uses random choice for the creation of child organisms.  
+
+```c++
+class Organism {
+public:
+    int genotype = 0;
+
+    Organism() {
+    }
+
+    Organism(int _genotype) {
+        genotype = _genotype;
+    }
+
+    int MutateGenotype(emp::Random &RandNum) {
+
+        double randMutation = RandNum.GetDouble(0, 1);
+
+        if (randMutation < mutRate) {
+            int MutatedGenotype = genotype - RandNum.GetInt(-3, 3);
+            genotype = MutatedGenotype;
+            cout << "mutated genotype = " << genotype << endl;
+        } else {
+            cout << "not mutated genotype = " << genotype << endl;
+        }
+
+        return genotype;
+    }
+};
+```
+
+**Pressure for Diversity**
+
+A goal of this project was to show that when we added a constraint that would incentivize the tree to branch more frequently, the overall diversity would increase. The way that this was accomplished was by adding a pressure for the tree to diversify by favoring rarer genotypes. 
```

Comment:
Maybe explain that the reason we had this goal was to show that our normalization technique can correctly identify a situation where there selective effects above and beyond those in the null model